packages/scheduled-tasks/src/lib/strategies/ScheduledTaskSQSStrategy.ts

```diff
@@ -0,0 +1,88 @@
+import { container } from '@sapphire/framework';
+import type { SQS } from 'aws-sdk';
+import { Consumer, ConsumerOptions } from 'sqs-consumer';
+import { Producer } from 'sqs-producer';
+import type { ScheduledTaskCreateRepeatedTask, ScheduledTasksTaskOptions } from '../types';
+import type { ScheduledTaskBaseStrategy } from '../types/ScheduledTaskBaseStrategy';
+
+export interface ScheduledTaskSQSStrategyMessageBody {
+	task: string;
+	payload?: any;
+	options: ScheduledTasksTaskOptions;
+}
+
+export class ScheduledTaskSQSStrategy implements ScheduledTaskBaseStrategy {
+	public readonly options: ConsumerOptions;
+
+	private producer!: Producer;
+
+	public constructor(options: ConsumerOptions) {
+		this.options = options;
+	}
+
+	public connect() {
+		this.producer = Producer.create(this.options);
+
+		const consumer = Consumer.create({
+			...this.options,
+			// handleMessage requires a promise method as value. But we don't need to..
+			// eslint-disable-next-line @typescript-eslint/require-await
+			handleMessage: async (message) => void this._handleMessage(message),
+			// eslint-disable-next-line @typescript-eslint/require-await
+			handleMessageBatch: async (messages) => void this._handleBatch(messages)
+		});
+		consumer.start();
+	}
+
+	public create(task: string, payload?: any, options?: ScheduledTasksTaskOptions) {
+		if (!this.producer) {
+			return;
+		}
+
+		if (options?.cron?.length) {
+			throw new Error('SQS does not support cron notation.');
+		}
+
+		let delay = options?.delay ? options?.delay / 1000 : 0;
+		if (options?.type === 'repeated') {
+			delay = options.interval! / 1000;
+		}
+
+		return this.producer.send({
+			id: `${task}-${Date.now()}`, // need it to be a unique ID'ish
```

Comment:
If you need unique id'ish.. either use something like crypto.randomBytes or nanoid, Date.now() CAN be non-unique